gpflow/utilities/utilities.py

```diff
@@ -39,6 +41,20 @@ def to_default_float(x):
     return tf.cast(x, dtype=default_float())
 
 
+class ShapeError(AssertionError):
+    pass
+
+
+def assert_shapes(shapes, *args, **kwargs):
```

Comment:
do we really need this wrapper? I, personally, like to be as close to tensorflow as possible.

Reply:
ti wants to use assertions to check code consistency, and I get that, but I don't think inconsistent code should throw an error that one might reaonsably catch. `tf.debugging.assert_shapes` throws a `ValueError` so we're converting it to throw a different error

Reply:
i agree we should be as close to tf as possible, but I think their functionality is limited here

Reply:
@joelberkeley-pio pointed out that ValueErrors are commonly try:except:ed and that this would hide the shape asserts. I think ideally TensorFlow's `assert_shapes` would raise a custom exception, not a ValueError; this wrapper makes clearer what went wrong. I think it'll make it a bit more user-friendly.

Can take it out again if that's the majority preference...